### PR TITLE
Make it possible to create orders via the admin

### PR DESCRIPTION
**Pull request by akx**

### CODE REVIEW

File: shoop/admin/modules/orders/json_order_creator.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+#
+# Copyright (c) 2012-2015, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+from __future__ import unicode_literals
+from copy import deepcopy
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
+from django.utils.translation import ugettext as _
+from shoop.core.models import (
```
Comment (tulimaki) [inline shoop/admin/modules/orders/json_order_creator.py]: Add blank line between django and shoop imports.

**Comment by tulimaki** [comment]
Most of the new files is missing Shoop license information.

**Comment by akx** [comment]
@tulimaki: I ran the license tool, isort and autopep8, but it seems that made your comments disappear. D:

**Comment by akx** [comment]
Oh, wait, some cache magic got me a snapshot:

image

**Comment by akx** [comment]
And to address the last two:

* Context: As far as I know, yes? Though in some cases, somehow, a request gets automagically converted into a context. Maybe @suutari-ai can clarify.
* Nowhere. We don't have a translation system in place for Javascript files.

File: shoop/admin/modules/orders/static_src/create/view/lines.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        return m("tr", [
+            m("td", selectBox(line.type, function () {
+                store.dispatch(setLineProperty(line.id, "type", this.value));
+            }, LINE_TYPES)),
+            m("td", line.sku),
+            m("td", {colspan: showPrice ? 1 : priceCells.length + 1}, text),
+            (showPrice ? priceCells : null),
+            m("td", m("button.btn.btn-xs.btn-danger", {
+                onclick: function () {
+                    store.dispatch(deleteLine(line.id));
```
Comment (Pikkupomo) [inline shoop/admin/modules/orders/static_src/create/view/lines.js]: `Uncaught TypeError: _actions.deleteLine is not a function` when clicking delete-button (trashcan). Points to this line

**Comment by Pikkupomo** [comment]
If I add variation parent into order, only that is added without possibility to select the variation child I actually want to add

**Comment by Pikkupomo** [comment]
* Order cannot be created if item has no supplier (product is not orderable). Error: `Could not create order: Order line has no supplier`. Error is in wrong color (blue) should be red.
* Tried to make discount-line, gave price of `-20` after saving and getting the error mentioned previously (supplier error) the price was 0 (not the original `-20`

**Comment by akx** [comment]
> * If I add variation parent into order, only that is added without possibility to select the variation child I actually want to add

Variation parents aren't orderable in the first place. There's currently no filter in the product browser to only choose from products that are actually orderable, and that's out of scope for this PR.

> * Order cannot be created if item has no supplier (product is not orderable).

As it should be.

> Error: `Could not create order: Order line has no supplier`. Error is in wrong color (blue) should be red.

I suppose the UX for errors should be much better in the first place (for instance, they could be shown on the related line), but that could be improved later.

> * Tried to make discount-line, gave price of `-20` after saving and getting the error mentioned previously (supplier error) the price was 0 (not the original `-20`

Discount lines weren't in the original spec at all.

File: shoop/admin/modules/orders/static_src/create/view/lines.js (hunk 2 above)
Comment (akx) [inline shoop/admin/modules/orders/static_src/create/view/lines.js]: Oops, yeah, was missing that action. Added.

**Pull request merged**